gnucash/gnome/assistant-stock-transaction.cpp

```diff
@@ -704,7 +742,30 @@ to ensure proper recording."), new_date_str, last_split_date_str);
                     NC_ ("Stock Assistant: Page name", "stock value"), errors);
 
 
-    if (info->txn_type->stock_amount != FieldMask::DISABLED)
+    if (info->txn_type->stock_amount == FieldMask::DISABLED)
+        ;
+    else if (info->txn_type->input_new_balance)
+    {
+        auto stock_amount = gnc_amount_edit_get_amount
+            (GNC_AMOUNT_EDIT(info->stock_amount_edit));
+        auto credit_side = (info->txn_type->stock_amount & FieldMask::ENABLED_CREDIT);
+        auto delta = gnc_numeric_sub_fixed (stock_amount, info->balance_at_date);
+        auto ratio = gnc_numeric_div (stock_amount, info->balance_at_date,
+                                      GNC_DENOM_AUTO, GNC_HOW_DENOM_REDUCE);
+        auto stock_pinfo = gnc_commodity_print_info
+            (xaccAccountGetCommodity (info->acct), true);
+        stock_amount = gnc_numeric_sub_fixed (stock_amount, info->balance_at_date);
+        line.units = xaccPrintAmount (stock_amount, stock_pinfo);
+        if (gnc_numeric_check (ratio))
+            add_error_str (errors, N_("Invalid stock new balance"));
```

Comment:
I do not understand this message. Perhaps a verb is missing?

Reply:
It means "The new stock balance that you entered is invalid because computing the split ratio from it caused an overflow, underflow, or divide by 0".

Reply:
Nearly. It can't be div/0 because info->balance_at_date is never zero in stock splits. It means the balance is garbage e.g."abc", overflow/underflow, or after 1752f1d19, zero or blank "".